### PR TITLE
Grant group KUBE_POD_LOG_READERS_GROUP access to read pod logs on gke…

### PR DESCRIPTION
**Pull request by vinayakankugoyal**
… control-plane.

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

All logs currently under /var/log/pods are only readable by root. This works as long as fluentbit is running as root. We are updating the master fluentbit component to run as non-root. Hence we need the logs to be readable by a group so that non root components can read them.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by loburm — APPROVED** [review]
/lgtm

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -1606,11 +1606,10 @@ EOF
 #
 # $1 is the file to create.
 # $2: the log owner uid to set for the log file.
-# $3: the log owner gid to set for the log file.
 function prepare-log-file {
   touch "$1"
-  chmod 644 "$1"
```
Comment (roycaihw) [inline cluster/gce/gci/configure-helper.sh:1612]: Will this break automations that read these logs as non-root?

nit: also, as a developer, now I need to sudo to read these logs when debugging in the master.

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -1606,11 +1606,10 @@ EOF
...
-  chown "${2:-${LOG_OWNER_USER:-root}}":"${3:-${LOG_OWNER_GROUP:-root}}" "$1"
+  chmod 640 "$1"
+  chown "${2:-root}":"${KUBE_POD_LOG_READERS_GROUP:-root}" "$1"
```
Comment (roycaihw) [inline cluster/gce/gci/configure-helper.sh:1612]: Is `KUBE_POD_LOG_READERS_GROUP` set on GCE? What does this change imply to GCE users? Shall we add a if condition and keep the old behavior if `KUBE_POD_LOG_READERS_GROUP` is not set?

File: cluster/gce/gci/configure-helper.sh [hunk 3]
```diff
@@ -1606,11 +1606,14 @@ EOF
 #
 # $1 is the file to create.
 # $2: the log owner uid to set for the log file.
-# $3: the log owner gid to set for the log file.
```
Comment (roycaihw) [inline cluster/gce/gci/configure-helper.sh:1609]: Keep this comment for the `KUBE_POD_LOG_READERS_GROUP` being empty case?

File: cluster/gce/gci/configure-helper.sh [hunk 4]
```diff
@@ -1606,11 +1606,14 @@ EOF
...
 function prepare-log-file {
   touch "$1"
   chmod 644 "$1"
-  chown "${2:-${LOG_OWNER_USER:-root}}":"${3:-${LOG_OWNER_GROUP:-root}}" "$1"
+  if [[ -n "${KUBE_POD_LOG_READERS_GROUP:-}" ]]; then
```
Comment (roycaihw) [inline cluster/gce/gci/configure-helper.sh:1612]: if `KUBE_POD_LOG_READERS_GROUP` is set, we can set 640 to unify the behavior across these logs and the pod logs.

File: cluster/gce/gci/configure-helper.sh (hunk 3 above)
Comment (vinayakankugoyal) [reply cluster/gce/gci/configure-helper.sh:1609]: Done.

File: cluster/gce/gci/configure-helper.sh (hunk 4 above)
Comment (vinayakankugoyal) [reply cluster/gce/gci/configure-helper.sh:1613]: Done.

**Pull request merged**